api/src/modules/scenarios/cost-surface/cost-surface.facade.ts

```diff
@@ -1,42 +1,44 @@
 import { Injectable } from '@nestjs/common';
-import { Request, Response } from 'express';
+import { Request } from 'express';
 import { AdjustCostSurface } from '../../analysis/entry-points/adjust-cost-surface';
-import { ProxyService } from '../../proxy/proxy.service';
+import { ResolvePuWithCost } from './resolve-pu-with-cost';
+import {
+  CostSurfaceEventsPort,
+  CostSurfaceState,
+} from './cost-surface-events.port';
+import { CostSurfaceInputDto } from '../../analysis/entry-points/adjust-cost-surface-input';
 
 @Injectable()
 export class CostSurfaceFacade {
   constructor(
     private readonly adjustCostSurfaceService: AdjustCostSurface,
-    private readonly proxyService: ProxyService,
+    private readonly shapefileConverter: ResolvePuWithCost,
+    private readonly events: CostSurfaceEventsPort,
   ) {}
 
   /**
    * non blocking - will do job in "background"
-   *
-   * @param scenarioId
-   * @param request
    */
-  convert(scenarioId: string, request: Request) {
-    // TODO #0 Generate & Dispatch Api Event (some wrapping service for /dummy/"terminating" if already running)
+  async convert(scenarioId: string, request: Request) {
+    await this.events.event(scenarioId, CostSurfaceState.Submitted);
+    let costSurface: CostSurfaceInputDto;
 
-    // TODO #1 Call Proxy Service to get Planning Units and their surface cost
-    // this.proxyService.... - modify this to send back data, not act on Response
+    try {
+      costSurface = await this.shapefileConverter.fromShapefile(request.file);
```

Comment:
The comment says it is not blocking (I understand it as the user should not await the returned promise). Actually, it is a bit – event dispatching is awaited (as well as a conversion), only the update below is not. Also, the inferred type of the method is `Promise`, and it's what a linter may complain about, and be misleading.

Theoretically, it may also result in an unhandled promise rejection.

What I would do is to just use `void` as a return type, and the entire `async` code I would put into `Promise.resolve().then(async () => {/*...*/}).catch(() => {/* swallowing/logging exception */})` or an async IIFE (`(async () => {/*...*/})().catch(() => {/* swallowing/logging exception */})`). It would allow to use normal `await`s while updating.

Reply:
or make it a normal async method and let the user of the method decide how to handle it

Reply:
Idea was to "you can, but you don't have to and won't get anything in return", thanks for pointing out missing piece!

Reply:
@Dyostiq https://github.com/Vizzuality/marxan-cloud/pull/196/commits/fef830aa4a16397bc40f76209dd3bf945f7db2c1